File: src/classes/Plant.java
package classes;

import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

/**
 * Created by pranavkonduru on 9/4/18.
 */
public final class Plant implements LifeForm {
    @NotNull
    @Override
    public Boolean move() {
        System.out.println("Plants cannot move");
        return false;
    }

    @NotNull
    @Override
    public String speak() {
        return "Plant Language";
    }

    @Nullable
    @Override
    public Integer limbs() {
        System.out.println("Plants do not have limbs");
        return null;
    }
}

File: src/classes/Plant.kt
package classes

/**
 * Created by pranavkonduru on 9/4/18.
 */
class Plant : LifeForm {

    override fun speak(): String {
        return "Plant Language"
    }

    override fun move(): Boolean {
        println("Plants cannot move")
        return false
    }

    override fun limbs(): Int? {
        println("Plants do not have limbs")
        return null
    }
}